echopype/convert/set_groups_azfp.py

```diff
@@ -147,24 +147,73 @@ def set_platform(self) -> xr.Dataset:
         unpacked_data = self.parser_obj.unpacked_data
         time2 = self.parser_obj.ping_time
 
+        # If tilt_x and/or tilt_y are all nan, create single-value timme2 dimension
+        # and single-value (np.nan) tilt_x and tilt_y
+        tilt_x = [np.nan] if np.isnan(unpacked_data["tilt_x"]).all() else unpacked_data["tilt_x"]
+        tilt_y = [np.nan] if np.isnan(unpacked_data["tilt_y"]).all() else unpacked_data["tilt_y"]
+        if (len(tilt_x) == 1 and np.isnan(tilt_x)) and (len(tilt_y) == 1 and np.isnan(tilt_y)):
+            time2 = [time2[0]]
+
         ds = xr.Dataset(
             {
+                "latitude": (
+                    ["time1"],
+                    [np.nan],
+                    self._varattrs["platform_var_default"]["latitude"],
+                ),
+                "longitude": (
+                    ["time1"],
+                    [np.nan],
+                    self._varattrs["platform_var_default"]["longitude"],
+                ),
+                "pitch": (
+                    ["time2"],
+                    [np.nan] * len(time2),
+                    self._varattrs["platform_var_default"]["pitch"],
+                ),
+                "roll": (
+                    ["time2"],
+                    [np.nan] * len(time2),
+                    self._varattrs["platform_var_default"]["roll"],
+                ),
+                "vertical_offset": (
+                    ["time2"],
+                    [np.nan] * len(time2),
+                    self._varattrs["platform_var_default"]["vertical_offset"],
+                ),
```

Comment:
> Are the variables separated out to be either with the time1 or the time2 dimension because we want to keep the consistency across sensors? 

Yes, that's the goal here.

> Also I wonder if it'll be useful to have time1=[time2[0]] somewhere before the dataset construction block, just for readability.

I like that. Maybe right below the statement `time2 = self.parser_obj.ping_time` (line 148), so that the data for all time dimensions are specified in one block?

Reply:
Done. Set `time1=[time2[0]]`.